core/src/ibc/transaction_handler/datagrams.rs

```diff
@@ -83,6 +85,11 @@ pub enum Datagram {
         proof_height: u64,
         consensus_height: u64,
     },
+    ConnOpenConfirm {
+        identifier: String,
+        proof_ack: Vec<u8>,
```

Comment:
How about `Bytes` for proofs and `ctypes::BlockNumber` for heights? I'll update previous ones (including this).

Reply:
I'll create a new commit that changes Vec<u8> to Bytes in all IBC code.
IMO, we should not use the ctypes::BlockNumber here. The should consider other chain's block number. `ctypes::BlockNumber` is for Foundry only.

Reply:
Ok then I'll leave `Vec<u8>`s. And please contain update to `ctypes::BlockNumber` for *our blocks*. (e.g *making* proof)

Reply:
How about defining a `IBCBlockNumber` type instead?
I'm not sure which method is better, but using a `IBCBlockNumber` type in the all IBC code is easier.

Reply:
Maybe together with `Identifier`